esphome/components/dashboard_import/__init__.py

```diff
@@ -105,7 +106,17 @@ def import_config(
             raise ValueError(f"Error while fetching {url}: {e}") from e
 
         contents = req.text
+        _, ind, bsi = load_yaml_guess_indent(contents)
+
         yaml = YAML()
+        yaml.preserve_quotes = True
+        yaml.width = (
+            max(
+                (len(contents_len) for contents_len in contents.splitlines()), default=0
+            )
+            + 10
+        )
+        yaml.indent(sequence=ind, offset=bsi)
```

Comment:
```suggestion
        yaml.indent(mapping=ind, sequence=ind, offset=bsi)
```

Do you need mapping as well?

Reply:
I think I've captured what you are solving in the tests, please check it